qxf2_scheduler/routes.py

```diff
@@ -65,103 +69,194 @@ def index():
+        Interviewers.interviewer_id == interviewer_id).values(Interviewers.interviewer_id,Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewers.interviewer_id, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+
+    parsed_interviewer_details = form_interviewer_details(interviewer_details)
+
+    return render_template("read-interviewers.html", result=parsed_interviewer_details)
+
+
+@app.route("/<interviewer_id>/interviewer/edit/", methods=['GET', 'POST'])
+def edit_interviewer(interviewer_id):
+    "Edit the interviewers"
+    # This query fetch the interviewer details by joining the time slots table and interviewers table.
+    edit_interviewer_details = Interviewers.query.join(Interviewertimeslots, Interviewers.interviewer_id == Interviewertimeslots.interviewer_id).values(
+        Interviewers.interviewer_id, Interviewers.interviewer_name, Interviewers.interviewer_email, Interviewers.interviewer_designation, Interviewertimeslots.interviewer_start_time, Interviewertimeslots.interviewer_end_time)
+
+    parsed_interviewer_details = form_interviewer_details(
+        edit_interviewer_details)
+
+    if request.method == "POST":
+        interviewer_name = request.form.get('name')
+        data = {'interviewer_name': interviewer_name}
+        print("ia interviewer is", interviewer_name, file=sys.stderr)
+        edit_interviewers = Interviewers.query.filter(Interviewers.interviewer_id == interviewer_id).update({'interviewer_name': request.form.get(
+            'name'), 'interviewer_email': request.form.get('email'), 'interviewer_designation': request.form.get('designation')})
+        db.session.commit()
+
+        # Adding the time slots in the interviewerstimeslots table
+        interviewer_time_slots = eval(request.form.get('timeObject'))
+        interviewer_start_time = interviewer_time_slots['starttime']
+        interviewer_end_time = interviewer_time_slots['endtime']
+
+        # Updating the table based on the time id of the interviewertimeslots table
+        # Fetching the time ids of interviewer
+        total_rows_of_interviewer_in_time_table = Interviewertimeslots.query.filter(
+            Interviewertimeslots.interviewer_id == interviewer_id).values(Interviewertimeslots.time_id)
+
+        # Updating the timeslots
+        i = 0
+        for each_time_id in total_rows_of_interviewer_in_time_table:
+            edited_time_slots = Interviewertimeslots.query.filter(Interviewertimeslots.time_id == each_time_id.time_id).update(
+                {'interviewer_start_time': interviewer_start_time[i], 'interviewer_end_time': interviewer_end_time[i]})
+            i = i + 1
+        db.session.commit()
+
+        return jsonify(data)
+
+    return render_template("edit-interviewer.html", result=parsed_interviewer_details)
+
+
+@app.route("/<interviewer_id>/interviewer/delete", methods=["POST"])
```

Comment:
The endpoint is always /entity/entity-id/verb. SO in this case, it has to be /interviewers/interviewer-id/delete

Reply:
Done